src/components/Legend/Legend.tsx

```diff
@@ -179,203 +103,113 @@ const StyledCardDescriptionSecond = styled(CardDescription)`
   width: 100%;
 `;
 
-const legendArray = [
-  {
-    label: '0',
-    value: 0,
-  },
-  {
-    label: '60',
-    value: 60,
-  },
-  {
-    label: '120',
-    value: 120,
-  },
-  {
-    label: '180',
-    value: 180,
-  },
-  {
-    label: '240',
-    value: 240,
-  },
-  {
-    label: '300',
-    value: 300,
-  },
-];
-
 const StyledToggle = styled.span`
   cursor: pointer;
   height: fit-content;
   &:hover {
     opacity: 0.66;
   }
 `;
-export interface LegendProps {
-  treesVisible: boolean;
-  rainVisible: boolean;
-  pumpsVisible: boolean;
-  legendExpanded: boolean;
-}
 
-function createCSSGradient(colors: string[], degrees = 90): string {
-  let gradient = `linear-gradient(${degrees}deg, `;
-  const len = colors.length;
-  let i = 0;
-  for (const color of colors) {
-    gradient += color;
-    if (i !== len - 1) {
-      gradient += ', ';
-    }
-    i++;
-  }
-  gradient += ')';
-  return gradient;
-}
-const rainColors = legendArray.map(item => interpolateColor(item.value));
+const rainColors = legendLabels.map(item => interpolateColor(item.value));
 const rainGradient = createCSSGradient(rainColors);
 
 const Legend: FC = () => {
-  const pumpsVisible = useStoreState('pumpsVisible');
-  const rainVisible = useStoreState('rainVisible');
-  const treesVisible = useStoreState('treesVisible');
+  const visibleLayer = useStoreState('visibleLayer');
+
   const [legendExpanded, setLegendExpanded] = useState<boolean>(true);
 
+  if (legendExpanded === false) {
+    return (
+      <LegendDiv>
+        <FlexSpace>
+          <FlexColumn>
+            <StyledCardDescription onClick={() => setLegendExpanded(true)}>
+              {'Legende'}
+            </StyledCardDescription>
+          </FlexColumn>
+          <StyledToggle onClick={() => setLegendExpanded(true)}>
+            {'+'}
+          </StyledToggle>
+        </FlexSpace>
+      </LegendDiv>
+    );
+  }
+
   return (
-    <LegendDiv active={legendExpanded}>
-      <FlexSpace active={legendExpanded}>
+    <LegendDiv active>
```

Comment:
Prefer `isActive` to `active` as it implies the use of a boolean more explicitly and makes the code more understandable.